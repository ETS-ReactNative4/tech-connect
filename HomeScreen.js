import React from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';
import { connect } from 'react-redux';
import { LinearGradient } from 'expo'
import { getUserInfo } from './thunks/getUserInfo'
import SuggestedConnection from './SuggestedConnection.js' 


export class HomeScreen extends React.Component {
  constructor(props) {
    super(props)
  }

  viewProfile = async (id) => {
    const user = await getUserInfo(id, this.props.user.api_key)
    this.props.navigation.navigate('ProfilePage', {user})
  }

  render() {
    const suggestions = this.props.user.suggestions
    return (
      <View style={styles.container}>
        <Text style={styles.name}>{this.props.user.name}</Text>
        <Text style={styles.position}>{this.props.user.position.job_title}</Text>
        <View style={styles.imageContainer}>
          <Image source={require('./profile-pic.jpeg')} style={styles.profilePic} />
        </View>
        <View style={styles.gradientContainer}>
          <LinearGradient
            colors={['#4AA9C5', '#6383C1', '#93548F']}
            start={[ 0.4, 0.7 ]}
            style={[styles.circle, { borderRadius: 300 }]} />
        </View>
        <View styles={styles.suggestedConnections}>
          <Text style={styles.suggestedTitle}>Suggested Connections</Text>
          { suggestions.map(suggestion => <SuggestedConnection suggestion={suggestion} viewProfile={this.viewProfile} />) }
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(HomeScreen)

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'flex-start',
    alignItems: 'stretch',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
    paddingTop: 20
  },
  gradientContainer: {
    shadowOffset: {  width: 0,  height: 2  },
    shadowRadius: 10,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  circle: {
    position: 'absolute',
    top: -690,
    left: -90,
    height: 600,
    width: 600
  },
  imageContainer: {
    display: 'flex',
    height: 180,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 200000,
    marginBottom: 20,
    shadowOffset: {  width: 5,  height: 5},
    shadowRadius: 5,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  profilePic: {
    height: 180,
    width: 180,
    borderRadius: 90,
    zIndex: 20000,
  },
  name: {
    marginTop: 20,
    marginLeft: 20,
    position: 'relative',
    zIndex: 20000,
    color: 'white',
    fontSize: 30
  },
  position: {
    marginTop: 5,
    marginLeft: 20,
    position: 'relative',
    zIndex: 20000,
    color: 'white',
    fontSize: 20,
  },
  suggestedTitle: {
    color: '#4AA9C5',
    fontSize: 20,
    marginBottom: 10,
    marginLeft: 20,
  },
  suggestedConnections: {
    display: 'flex',
    alignItems: 'stretch',
    shadowOffset: {  width: 0,  height: 2 },
    shadowRadius: 10,
    shadowColor: 'black',
    shadowOpacity: .5,
    width: '100%',
    backgroundColor: 'blue'
  }
})




