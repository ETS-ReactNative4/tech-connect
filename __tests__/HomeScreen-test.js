import React from 'react';
import { HomeScreen, mapStateToProps } from '../HomeScreen';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('HomeScreen', () => {
  let wrapper

  beforeEach(() => {
    mockUser = {
      name: 'Howard',
      position: {
        job_title: 'astronomer'
      }, 
      suggestions: [],
      api_key: 112345129372873,
    }
    wrapper = shallow(<HomeScreen user={ mockUser } />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })
})