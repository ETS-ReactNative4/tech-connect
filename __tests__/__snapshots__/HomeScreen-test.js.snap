// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match the snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "marginLeft": 20,
        "marginTop": 20,
        "position": "relative",
        "zIndex": 20000,
      }
    }
  >
    Howard
  </Component>
  <Component
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "marginLeft": 20,
        "marginTop": 5,
        "position": "relative",
        "zIndex": 20000,
      }
    }
  >
    astronomer
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 180,
        "justifyContent": "center",
        "marginBottom": 20,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
        "width": "100%",
        "zIndex": 200000,
      }
    }
  >
    <Component
      source={1}
      style={
        Object {
          "borderRadius": 90,
          "height": 180,
          "width": 180,
          "zIndex": 20000,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 10,
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "#4AA9C5",
          "#6383C1",
          "#93548F",
        ]
      }
      start={
        Array [
          0.4,
          0.7,
        ]
      }
      style={
        Array [
          Object {
            "height": 600,
            "left": -90,
            "position": "absolute",
            "top": -690,
            "width": 600,
          },
          Object {
            "borderRadius": 300,
          },
        ]
      }
    />
  </Component>
  <Component
    styles={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "blue",
        "display": "flex",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 10,
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#4AA9C5",
          "fontSize": 20,
          "marginBottom": 10,
          "marginLeft": 20,
        }
      }
    >
      Suggested Connections
    </Component>
  </Component>
</Component>
`;
